### PR TITLE
Upgrade global styles

### PR DESCRIPTION
**Pull request by zchsh**
:tickets: [Asana ticket](https://app.asana.com/0/1100423001970639/1199185162145231/f)
:mag: [Deploy preview](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app)

This PR upgrades to the latest `@hashicorp/mtkg-global-styles` release. This release cuts down on the amount of global CSS we ship as part of that package. This PR also upgrades to new versions of related components, in particular `@hashicorp/react-content`, which is used on all docs pages.

### Single page views

- [/](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/) vs [live](https://www.consul.io/)
- [/home](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/home) vs [live](https://www.consul.io/home)
  - note: duplicate of `/`, we have [a ticket to address this](https://app.asana.com/0/1100423001970639/1199638081194663/f)
- [/404](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/404) vs [live](https://www.consul.io/404)
- [/community](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/community) vs [live](https://www.consul.io/community)
- [/downloads](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/downloads) vs [live](https://www.consul.io/downloads)
- [/security](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/security) vs [live](https://www.consul.io/security)
- [/use-cases/multi-platform-service-mesh](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/use-cases/multi-platform-service-mesh) vs [live](https://www.consul.io/use-cases/multi-platform-service-mesh)
- [/use-cases/network-infrastructure-automation](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/use-cases/network-infrastructure-automation) vs [live](https://www.consul.io/use-cases/network-infrastructure-automation)
- [/use-cases/service-discovery-and-health-checking](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/use-cases/service-discovery-and-health-checking) vs [live](https://www.consul.io/use-cases/service-discovery-and-health-checking)

### Docs views

All views that use `docs-page` use an updated `content` component with localized `table` styles, as those styles are no longer shipped as global CSS. The updated `docs-page` component also contains minor alignment fixes.

- [/docs/[[...page]]](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/docs/) vs [live](https://www.consul.io/docs/)
- [/api-docs/[[...page]]](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/api-docs/) vs [live](https://www.consul.io/api-docs/)
- [/commands/[[...page]]](https://consul-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/commands/) vs [live](https://www.consul.io/commands/)

### CODE REVIEW

File: website/components/callout-blade/index.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default function CalloutBlade({ title, callouts }) {
   return (
     <div className={styles.calloutBlade}>
       <div className={styles.contentWrapper}>
-        <h3>{title}</h3>
+        <h3 className="g-type-display-3">{title}</h3>
```
Comment (zchsh) [inline website/components/callout-blade/index.jsx:10]: Backfilled missing type styles.

File: website/components/footer/index.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Link from 'next/link'
 export default function Footer({ openConsentManager }) {
   return (
     <footer className="g-footer">
-      <div className="g-container">
+      <div className="g-grid-container">
```
Comment (zchsh) [inline website/components/footer/index.jsx:6]: Ran replace of `g-container` - here in footer seems to be the only instance.

File: website/pages/_app.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ export default function App({ Component, pageProps }) {
         />
       </HashiHead>
       {ALERT_BANNER_ACTIVE && (
-        <AlertBanner {...alertBannerData} product="consul" />
+        <AlertBanner {...alertBannerData} product="consul" hideOnMobile />
```
Comment (zchsh) [inline website/pages/_app.js:42]: To match previous behaviour due to global styles previously shipped as part of `docs-page`, alert-banner is hidden on mobile.

File: website/pages/community/index.jsx [hunk 4]
```diff
@@ -2,10 +2,11 @@ import VerticalTextBlockList from '@hashicorp/react-vertical-text-block-list'
 import SectionHeader from '@hashicorp/react-section-header'
 import Head from 'next/head'
 import { productSlug } from 'data/metadata'
+import s from './style.module.css'
 
 export default function CommunityPage() {
   return (
-    <div id="p-community">
+    <div className={s.root}>
```
Comment (zchsh) [inline website/pages/community/index.jsx:9]: Needed to remove deprecated `--site-max-width`, was straightforward to convert the few lines of CSS to CSS modules.

File: website/pages/downloads/style.module.css [hunk 5]
```diff
@@ -8,6 +8,10 @@
   border-radius: 3px;
   margin: 24px 0;
 
+  & code {
+    font-size: 0.8125rem;
+  }
+
```
Comment (zchsh) [inline website/pages/downloads/style.module.css:14]: Minor fix since code font-size is no longer set in global styles.

**Review by BRKalow — APPROVED** [review]
👏

**Review by jescalan — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/431376.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit c42ea82883ffe78bb06f7fdd137ad7fcb7adef59 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit c42ea82883ffe78bb06f7fdd137ad7fcb7adef59 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/431376)

**Pull request merged**